internal/venice-common/src/main/java/com/linkedin/venice/ConfigKeys.java

```diff
@@ -731,6 +731,34 @@ private ConfigKeys() {
   public static final String SERVER_USE_METRICS_BASED_POSITION_IN_LAG_COMPUTATION =
       "server.use.metrics.based.position.in.lag.computation";
 
+  /**
+   * Controls whether to use upstreamPubSubPosition (with offset fallback) or upstreamOffset
+   * when deserializing positions from leader metadata during ingestion.
+   *
+   * Default: true (use upstreamPubSubPosition with fallback to upstreamOffset).
+   * When true, the server will attempt to deserialize the position from upstreamPubSubPosition
+   * and fall back to upstreamOffset if deserialization fails or the position is invalid.
+   * When false, the server will directly use upstreamOffset without attempting position deserialization.
+   *
+   * This provides a safety mechanism to revert to offset-only behavior if position deserialization
+   * causes issues in production.
+   */
+  public static final String SERVER_USE_UPSTREAM_PUBSUB_POSITION_WITH_FALLBACK =
+      "server.use.upstream.pubsub.position.with.fallback";
```

Comment:
```suggestion
      "server.upstream.position.fallback.enabled";
```
How about a shorter name like this? Same with `checkpointed`.